pyaedt/circuit.py

```diff
@@ -1431,6 +1431,90 @@ def assign_power_sinusoidal_excitation_to_ports(self, ports, settings):
 
         return True
 
+    @pyaedt_function_handler()
+    def assign_voltage_frequency_dependent_excitation_to_ports(self, ports, filepath):
+        """Assign a frequency dependent excitation to circuit ports from a frequency dependent source (.fds format).
+
+        Parameters
+        ----------
+        ports : list
+            List of circuit ports to assign to the frequency dependent excitation.
+        filepath : str
+            Path to the frequency dependent file.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+
+        >>> oDesign.UpdateSources
+        """
+        id = self.modeler.schematic.create_unique_id()
```

Comment:
maybe it's worth to add a check if the file path provided exists and then another check that the extension of the file name provided is valid and is .fds format.
You could also add a unit test for that and check that it fails if file doesn't exist or file extension is wrong.

Reply:
@gmalinve Done